### DISCUSSION TITLE
5 functions that take 30 seconds, split or keep together?

### DISCUSSION THREAD

**Discussion by notflip** [hammerstonedev/sidecar#115] [Q&A]
I'm building a scraper that scrapes 5 different websites for information, now I'm wondering

- Is it better to create 5 separate sidecar functions, so 5 lambda functions, update the client of the progress after each step
- Or create 1 big function that performs all 5 actions, which means I won't be able to update the client on the progress as easily

What is the best practise?

**Comment by gcavanunez** [comment]
Hey, so thinking in terms of fault tolerance, if they truly don't depend on one another, way better to separate them into 5 processes. 

That being said, I'm noticing there is not discussions tab on the repo, I'm wondering whether that is intentional or somehting worth opening up to funnel these types of conversations.

**Comment by aarondfrancis** [comment]
Might be better to have a discussion tab, because this is kind of outside the context of an issue on this particular repo. I would also agree that splitting them up would be ideal!
